src/compile/facet.ts

```diff
@@ -332,6 +332,9 @@ function getFacetGroupProperties(model: FacetModel) {
   const encodeEntry = model.assembleParentGroupProperties();
 
   return {
+    width: model.child.getSizeSignalRef('width'),
```

Comment:
I wonder if this is the right fix.  Shouldn't `model.assembleParentGroupProperties` already produce what we need?  

I wonder if there is something actually wrong with `hasSubPlotWithXy`? 
(As commented, I'm not sure if we really need it anyway.)